### PR TITLE
Add Boolean support for the PersistentDataStorage.

### PR DESCRIPTION
**Pull request by zenker**

### CODE REVIEW

**Comment by zenker** [comment]
I left the test almost untouched and just added ChimeraTK::Boolean. One should test all types via templated test case...

**Comment by zenker** [comment]
Maybe one should increase the minor version rather than the patch version. It's up to @mhier.

**Comment by killenb** [comment]
> Maybe one should increase the minor version rather than the patch version. It's up to @mhier.

Yes, it should be a minor version increase because a feature was added. Servers which use the bool cannot use an older version of the library.

**Comment by mhier** [comment]
> > Maybe one should increase the minor version rather than the patch version. It's up to @mhier.
> 
> Yes, it should be a minor version increase because a feature was added. Servers which use the bool cannot use an older version of the library.

Well it depends how you view it: In a sense it can be considered a bug that it was forgotten to implement this feature when adding support for the new DeviceAccess version.

Technically it is acceptable to make it a patch release, since no header file was touched (binary compatibility is there). The advantage of a patch version is that you won't need to recompile applications.

So if you prefer you can keep it a patch release from my point of view.

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@ include(cmake/add_dependency.cmake)
 
 set(${PROJECT_NAME}_MAJOR_VERSION 02)
 set(${PROJECT_NAME}_MINOR_VERSION 03)
-set(${PROJECT_NAME}_PATCH_VERSION 00)
+set(${PROJECT_NAME}_PATCH_VERSION 01)
```
Comment (mhier) [inline CMakeLists.txt:9]: See discussion in the Conversation tab...

**Review by mhier — APPROVED** [review]

**Comment by mhier** [comment]
@killenb: If you are ok with a patch release, please merge it.

**Pull request merged**